packages/integrations/components/pico/observer.js

```diff
@@ -0,0 +1,200 @@
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import debounce from 'lodash/debounce';
+import useMutationObserver from '../../hooks/useMutationObserver';
+import {
+  actionVerifyPicoUser,
+  actionRequireUpgrade,
+  actionReceivePicoPlanUpgrade,
+} from '../../actions/picoActions';
+import {
+  showUpgradeModalSelector,
+  upgradeModalDismissedSelector,
+} from '../../selectors/coralSelector';
+
+const PicoObserver = ({
+  tiers,
+  accessToken,
+  logoutRequestAction,
+}) => {
+  // Declare state variables for effects to be run against.
+  const [email, setEmail] = useState(undefined);
+  const [isAnonymous, setIsAnonymous] = useState(false);
+  const [isRegistered, setIsRegistered] = useState(false);
+  const [isPaying, setIsPaying] = useState(false);
+  const [canComment, setCanComment] = useState(false);
+
+  const manageLocalState = (target) => {
+    const emailAddress = target.getAttribute('data-pico-email');
+    const status = target.getAttribute('data-pico-status');
+    const tier = target.getAttribute('data-pico-tier');
+
+    const hasValidStatus = 'registered' === status || 'paying' === status;
+    const hasNotUpdated = ! isAnonymous || ! isRegistered || ! isPaying;
+
+    if ('excluded' !== status && hasValidStatus && hasNotUpdated) {
+      // Update the `email` value with the user's email address.
+      if (undefined === email) {
+        setEmail(emailAddress);
+      }
+      // If the user's tier is included in the whitelisted tiers for SSO,
+      // flag their ability to comment as true.
+      if (tiers.includes(tier)) {
+        setCanComment(true);
+      }
+      // If the status is registered, set the isRegistered flag to true.
+      if ('registered' === status) {
+        setIsRegistered(true);
+      }
+      // If the status is paying, set the isPaying flag to true.
+      if ('paying' === status) {
+        setIsPaying(true);
+      }
+    }
+
+    // If the user was previously recognized as registered or paying, but
+    // is then anonymized, update the state values accordingly.
+    if ((isRegistered || isPaying) && 'anonymous' === status) {
```

Comment:
Is there a case where `isPaying` would be true, but `isRegistered` would not be?

Reply:
Yes. Pico uses `registered` and `paying` as separate identifiers in the `data-pico-status` attribute. If a user is registered with Pico, but has not subscribed, they will be marked as registered. If they are paying, regardless of tier, they'll simply be marked as paying.